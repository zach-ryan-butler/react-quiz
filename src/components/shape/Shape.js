import React from 'react';
import PropTypes from 'prop-types';

function Shape({ text, color, backgroundColor }) {
  return (
    <section style={{ backgroundColor, height: '100px', width: '100px' }} >
      <p style={{ color }} >{text}</p>
    </section>
  );
}

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string,
  backgroundColor: PropTypes.string
};

export default Shape;
