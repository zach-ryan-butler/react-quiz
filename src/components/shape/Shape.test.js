import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders Shape', () => {
    const wrapper = shallow(<Shape
      text="Zach Butler" 
      color="red"
      backgroundColor="red" />);
    expect(wrapper).toMatchSnapshot();
  });
});
