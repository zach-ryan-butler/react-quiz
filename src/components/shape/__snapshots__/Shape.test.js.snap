// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders Shape 1`] = `
<section
  style={
    Object {
      "backgroundColor": "red",
      "height": "100px",
      "width": "100px",
    }
  }
>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Zach Butler
  </p>
</section>
`;

exports[`Shape component renders TextSelector 1`] = `
<section
  style={
    Object {
      "backgroundColor": "red",
      "height": "100px",
      "width": "100px",
    }
  }
>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Zach Butler
  </p>
</section>
`;

exports[`TextSelector component renders TextSelector 1`] = `
<section
  style={
    Object {
      "backgroundColor": "red",
      "height": "100px",
      "width": "100px",
    }
  }
>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  >
    Zach Butler
  </p>
</section>
`;
